src/resources/loader.js

```diff
@@ -76,47 +76,41 @@ Object.assign(pc, function () {
                 // new request
                 this._requests[key] = [callback];
 
+                var loader = this;
                 var handleLoad = function (err, urlObj) {
                     if (err) {
                         console.error(err);
-                        if (this._requests[key]) {
-                            for (var i = 0, len = this._requests[key].length; i < len; i++) {
-                                this._requests[key][i](err);
+                        if (loader._requests[key]) {
+                            for (var i = 0, len = loader._requests[key].length; i < len; i++) {
+                                loader._requests[key][i](err);
                             }
                         }
-                        delete this._requests[key];
+                        delete loader._requests[key];
                         return;
                     }
 
                     handler.load(urlObj, function (err, data, extra) {
                         // make sure key exists because loader
                         // might have been destroyed by now
-                        if (!this._requests[key])
+                        if (!loader._requests[key])
                             return;
 
-                        var i, len = this._requests[key].length;
-
-                        var resource;
-                        if (! err) {
+                        if (!err) {
                             try {
-                                resource = handler.open(urlObj.original, data, asset);
+                                if (!handler.openAsync ||
+                                    !handler.openAsync(urlObj.original, data, asset,
```

Comment:
Is openAsync expected to return a value? Looks like it doesn't need to.

Reply:
As an aside from this. It think we should look into (as a separate feature/PR) changing the handlers so that they are all async. I'm not sure there is a strong reason to try and support two code paths and it complicates the code.

Reply:
1. There is a special case - some one can add custom model parser (it is a public API) that parser will only support sync loading (no callback, etc.) for the specific file extension. And the handler async loading will not find a parser, so the it has to fall back to sync open.

Reply:
2. Overall the code will greatly benefit from this but we need to decide if we break user custom parsers (if there are any).

Reply:
Re: 2. Honestly I think breaking external model parsers is probably fine. The only one I know about is the OBJ one that I wrote.